compiler/cpp/src/thrift/generate/t_rs_generator.cc

```diff
@@ -673,15 +676,22 @@ void t_rs_generator::render_const_value_holder(const string& name, t_type* ttype
   f_gen_ << endl;
 }
 
-void t_rs_generator::render_const_value(t_type* ttype, t_const_value* tvalue) {
+void t_rs_generator::render_const_value(t_type* ttype, t_const_value* tvalue, bool isowned) {
   if (ttype->is_base_type()) {
     t_base_type* tbase_type = (t_base_type*)ttype;
     switch (tbase_type->get_base()) {
     case t_base_type::TYPE_STRING:
       if (tbase_type->is_binary()) {
-        f_gen_ << "\"" << tvalue->get_string() << "\""<<  ".to_owned().into_bytes()";
+        if (isowned) {
```

Comment:
Will there ever be a point where we want the const string/byte-array to be owned?

Reply:
Honestly, I'm not really sure about the byte-arrays.  We aren't using them anywhere, so I'm not clear on the use-cases.
I assume it will end up like the regular strings where you might want `map<u8,binary>` or something like that.

Reply:
My goals were:
1. Make `const binary` compile
2. Preserve existing behavior

Presently, neither `const string` nor `const binary` compiles. I'm more interested in getting our stuff compiling than addressing the semantics of `binary`. Should I revert the binary portion?

Reply:
No - the PR isn't being held up because of const binary. I just didn't have time to look at it yet. I usually pull the branch locally and test in a container to ensure everything works. For one thing, I'm unsure that

```
const FOO: &str = "ladeda";
```

works in rust 1.28.

EDIT. Oh, nm - I see it was [enabled in 1.17](https://rust-lang-nursery.github.io/edition-guide/rust-2018/ownership-and-lifetimes/simpler-lifetimes-in-static-and-const.html)

Reply:
That explains it.  I could have sworn it needed to be `&'static`, but I found a bunch of examples that were `&str` and assumed I was thinking of something else...
Do you think it's worth outputting `&'static str` instead so it works with Rust <1.17?